employee-rostering-frontend/src/store/employee/employee.test.ts

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { mockStore } from '../mockStore';
+import { AppState } from '../types';
+import * as actions from './actions';
+import reducer, { employeeOperations } from './index';
+import {withElement, withoutElement, withUpdatedElement} from 'util/ImmutableCollectionOperations';
+import {onGet, onPost, onDelete, resetRestClientMock} from 'store/rest/RestTestUtils';
+import Employee from 'domain/Employee';
+
+describe('Employee operations', () => {
+  it('should dispatch actions and call client', async () => {
+    const { store, client } = mockStore(state);
+    const tenantId = store.getState().tenantData.currentTenantId;
+    
+    const mockEmployee: Employee = {
+      tenantId: tenantId,
+      id: 0,
+      version: 0,
+      name: "Employee 1",
+      skillProficiencySet: [],
+      contract: {
+        tenantId: tenantId,
+        id: 1,
+        name: "Contract",
+        maximumMinutesPerDay: null,
+        maximumMinutesPerWeek: null,
+        maximumMinutesPerMonth: null,
+        maximumMinutesPerYear: null
+      }
+    };
+
+    const mockEmployeeList: Employee[] = [mockEmployee];
+
+    onGet(`/tenant/${tenantId}/employee/`, mockEmployeeList);
+    await store.dispatch(employeeOperations.refreshEmployeeList());
+    expect(store.getActions()).toEqual([actions.refreshEmployeeList(mockEmployeeList)]);
+    expect(client.get).toHaveBeenCalledTimes(1);
+    expect(client.get).toHaveBeenCalledWith(`/tenant/${tenantId}/employee/`);
+
+    store.clearActions();
+    resetRestClientMock(client);
```

Comment:
I suggest to split the test so that each use case (get/delete/post etc.) has its own test method. This will make the test more readable and isolated so it's not necessary to this kind of clean up.

Reply:
I modelled it after the `optaweb-vehicle-routing` Routes test (which have all operations in a single test): https://github.com/kiegroup/optaweb-vehicle-routing/blob/master/optaweb-vehicle-routing-frontend/src/store/route/route.test.ts. Probably best to update that one too.

Reply:
Done: kiegroup/optaweb-vehicle-routing/pull/126.